dex-load/src/main/scala/com/wavesplatform/dex/load/TankGenerator.scala

```diff
@@ -0,0 +1,215 @@
+    val pairs    = readPairs(pairsFile)
+    val ts       = System.currentTimeMillis
+
+    val all = accounts.flatMap(a => {
+      Request(
+        Request.GET.toString,
+        s"/matcher/balance/reserved/${Base58.encode(a.getPublicKey())}",
+        Request.RESERVED_BALANCE.toString,
+        headers = Map("Signature" -> settings.matcher.getOrderHistorySignature(a, ts), "Timestamp" -> ts.toString)
+      ) ::
+        pairs.map(p => {
+        Request(
+          Request.GET.toString,
+          s"/matcher/orderbook/${p.getAmountAsset}/${p.getPriceAsset}/tradableBalance/${a.getAddress}",
+          Request.TRADABLE_BALANCE.toString,
+          headers = Map("Signature" -> settings.matcher.getOrderHistorySignature(a, ts), "Timestamp" -> ts.toString)
+        )
+      })
+    })
+
+    Random.shuffle(List.fill(1000)(all).flatten.take(requestsCount)) //TODO: calculate needed count
+  }
+
+  private def mkAllTypes(seedPrefix: String, requestsCount: Int = 20000): List[Request] = {
+    println("Making requests:")
+    Random.shuffle(
+      mkMatching(seedPrefix, requestsCount) ++ mkBalances(seedPrefix, requestsCount, new File("pairs.txt")) ++
+        mkOrderHistory(seedPrefix, requestsCount, new File("pairs.txt"))
+    )
+  }
+
+  private def svRequests(requests: List[Request], outputFile: File): Unit = {
+    println("\nAll data has been generated. Now it will be saved...")
+
+    val output = new PrintWriter(outputFile, "utf-8")
+
+    try requests.foreach(_.save(output))
+    finally output.close()
+
+    println(s"Generated: ${requests.length}")
+    println(s"\t${Request.POST}: ${requests.count(_.httpType.equals("POST"))}")
+    println(s"\t\t${Request.PLACE}: ${requests.filter(_.tag.equals("PLACE")).length}")
+    println(s"\t\t${Request.CANCEL}: ${requests.filter(_.tag.equals("CANCEL")).length}")
+    println(s"\t${Request.GET}: ${requests.filter(_.httpType.equals("GET")).length}")
+    println(s"\t\t${Request.ORDER_HISTORY_BY_ACC}: ${requests.filter(_.tag.equals("ORDER_HISTORY_BY_ACC")).length}")
+    println(s"\t\t${Request.ORDER_HISTORY_BY_PAIR}: ${requests.filter(_.tag.equals("ORDER_HISTORY_BY_PAIR")).length}")
+    println(s"\t\t${Request.RESERVED_BALANCE}: ${requests.filter(_.tag.equals("RESERVED_BALANCE")).length}")
+    println(s"\t\t${Request.TRADABLE_BALANCE}: ${requests.filter(_.tag.equals("TRADABLE_BALANCE")).length}")
+    println(s"Results have been saved to $outputFile")
+  }
+
+  def mkRequests(seedPrefix: String, pairsFile: Option[File], outputFile: File, requestsCount: Int = 20000, requestsType: Int = 3): Unit = {
+    val requests = requestsType match {
+      case 1 => mkPlaces(seedPrefix, requestsCount)
+      case 2 => mkCancels(seedPrefix, requestsCount)
+      case 3 => mkMatching(seedPrefix, requestsCount)
+      case 4 => mkOrderHistory(seedPrefix, requestsCount, pairsFile.get)
+      case 5 => mkBalances(seedPrefix, requestsCount, pairsFile.get)
+      case 6 => mkAllTypes(seedPrefix, requestsCount)
```

Comment:
This is bloody unexpected that some of these methods generate new assets instead of using `pairsFile`.
Consider to rewrite `pairsFile` in request types that generate assets too.

Reply:
fixed

Reply:
I set up this argument as nonrequired. If the user didn't pass `-pf` assets will be generated and pairs will be randomly formed.